src/Brahma.FSharp.OpenCL.Printer/FunDecl.fs

```diff
@@ -20,43 +20,53 @@ open Microsoft.FSharp.Text.StructuredFormat
 open Microsoft.FSharp.Text.StructuredFormat.LayoutOps
 open Brahma.FSharp.OpenCL.Printer
 
-let private printFunFormalParam (param:FunFormalArg<_>) =
-    [ match param.DeclSpecs.AddressSpaceQual with
-      | Global ->
-           if param.Name.[0] <> '_'
-           then yield wordL "__global"
-           else yield wordL "__global"
-      | _ -> yield wordL "private"
-    ; match param.DeclSpecs.Type with
-      | Some t -> yield Types.Print t
-      | None -> failwith "Could not print a formal arg with undefined type"
-    ; yield wordL param.Name
-    ] |> spaceListL
-
-let Print<'lang> (funDecl:FunDecl<'lang>) =
+let private printFunFormalParam (param: FunFormalArg<_>) =
+    [
+        match param.DeclSpecs.AddressSpaceQualifier with
+        | Global ->
+            // TODO wtf: equal branches
+            if param.Name.[0] <> '_' then
```

Comment:
I guess, it can be cleaned up. Looks like an attempt to hack the memory modifiers.